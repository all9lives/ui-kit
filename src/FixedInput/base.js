import { css } from 'styled-components'
import Color from '../utils/Color'

const base = css`
  background-color: ${Color.white};
`

export default base
