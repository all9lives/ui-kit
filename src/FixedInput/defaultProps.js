export default {
  placeholder: 'Placeholder',
  saveText: 'Save',
  label: 'Label'
}
