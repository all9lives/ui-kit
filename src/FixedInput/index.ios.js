import FixedInput from './index.android'

export default FixedInput
