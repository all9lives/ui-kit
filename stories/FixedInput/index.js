import React, { Component, cloneElement } from 'react'
import { action } from '@storybook/addon-actions'
import createStory from '../../storybook/createStory'
import FixedInput from '../../src/FixedInput'
import View from '../../src/View'
import Button from '../../src/Button'

class StateProvider extends Component {
  state = {
    show: false
  }
  render () {
    return (
      <View style={{ width: '100%', height: '100%', paddingTop: 30 }} align='center'>
        <Button onPress={() => this.setState({ show: true })} title='Edit' />
        {cloneElement(
          this.props.children,
          {
            onRequestDismiss: () => {
              if (this.props.onRequestDismiss) this.props.onRequestDismiss()
              this.setState({ show: false })
            },
            show: this.state.show
          }
        )}
      </View>
    )
  }
}

createStory('FixedInput', module)
  .add('default', () => (
    <StateProvider>
      <FixedInput />
    </StateProvider>
  ))
  .add('with multiline', () => (
    <StateProvider>
      <FixedInput multiline />
    </StateProvider>
  ))
  .add('with label', () => (
    <StateProvider>
      <FixedInput label='MyLabel' />
    </StateProvider>
  ))
  .add('with saveText', () => (
    <StateProvider>
      <FixedInput saveText='MySaveText' />
    </StateProvider>
  ))
  .add('with onRequestDismiss', () => (
    <StateProvider onRequestDismiss={action('onRequestDismiss')}>
      <FixedInput />
    </StateProvider>
  ))
  .add('with onChangeText', () => (
    <StateProvider>
      <FixedInput onChangeText={action('Input onChangeText')} />
    </StateProvider>
  ))
