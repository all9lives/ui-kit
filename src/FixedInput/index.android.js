import React, { Component } from 'react'
import { Modal, Keyboard, KeyboardAvoidingView, TouchableWithoutFeedback } from 'react-native'
import styled from 'styled-components/native'
import base from './base'
import defaultProps from './defaultProps'
import Unit from '../utils/Unit'
import Color from '../utils/Color'
import Size from '../utils/Size'
import View from '../View'
import Text from '../Text'
import Button from '../Button'
import Input from '../Input'

const Container = styled(View)`
  ${base}
  width: 100%;
  position: absolute;
  bottom: 0;
  left: 0;
`

const Header = styled(View)`
  border-bottom-width: 1px;
  border-bottom-color: ${Color.gray};
  padding: ${Unit.convert(Size.spacing * 2)} ${Unit.convert(Size.spacing)};
`

const Save = styled(View)`
  position: absolute;
  top: ${Unit.convert(Size.spacing)};
  right: ${Unit.convert(Size.spacing)};
`

const Body = styled(View)`
  padding: ${Unit.convert(Size.spacing)} ${Unit.convert(Size.spacing)} ${Unit.convert(Size.spacing * 2)};
`

export default class FixedInput extends Component {
  static defaultProps = defaultProps
  constructor (props) {
    super(props)
    this.state = {
      value: props.value
    }
  }
  componentDidMount () {
    this.keyboardDidHideListener = Keyboard.addListener('keyboardDidHide', this._keyboardDidHide)
  }
  componentDidUpdate (prevProps) {
    if (this.props.show !== prevProps.show) {
      this.setState({ value: this.props.show ? this.props.value : '' })
    }
  }
  componentWillUnmount () {
    this.keyboardDidHideListener.remove()
  }
  _keyboardDidHide = () => {
    this.props.onRequestDismiss()
  }
  handlePress = () => {
    if (this.props.onChangeText) {
      this.props.onChangeText(this.state.value)
    }
  }
  handleChangeText = (value) => {
    this.setState({ value })
  }
  render () {
    const { show, label, saveText, onRequestDismiss, value, ...props } = this.props
    if (!show) {
      return null
    }
    return (
      <Modal visible transparent onRequestClose={onRequestDismiss}>
        <KeyboardAvoidingView style={{ flex: 1 }} behavior='height'>
          <TouchableWithoutFeedback onPress={onRequestDismiss}>
            <View grow={1} />
          </TouchableWithoutFeedback>
          <Container>
            <Header align='center' direction='row'>
              <Text>{label}</Text>
              <Save>
                <Button title={saveText} color='gray' onPress={this.handlePress} />
              </Save>
            </Header>
            <Body>
              <Input {...props} value={this.state.value || value} autoFocus onChangeText={this.handleChangeText} />
            </Body>
          </Container>
        </KeyboardAvoidingView>
      </Modal>
    )
  }
}
