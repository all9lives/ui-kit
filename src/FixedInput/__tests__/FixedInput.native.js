/* eslint-env jest */
import React from 'react'
import { shallow } from 'enzyme'
import FixedInput from '../index.ios.js'

describe('FixedInput.native', () => {
  it('renders a snapshot', () => {
    const wrapper = shallow(
      <FixedInput />
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('renders a snapshot with show', () => {
    const wrapper = shallow(
      <FixedInput show />
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('renders a snapshot with show and multiline', () => {
    const wrapper = shallow(
      <FixedInput show multiline />
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('renders a snapshot with show and label', () => {
    const wrapper = shallow(
      <FixedInput show label='MyLabel' />
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('renders a snapshot with show and saveText', () => {
    const wrapper = shallow(
      <FixedInput show saveText='MySaveText' />
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('renders a snapshot with show and onRequestDismiss', () => {
    const wrapper = shallow(
      <FixedInput show onRequestDismiss={() => console.log('dismiss')} />
    )
    expect(wrapper).toMatchSnapshot()
  })
})
