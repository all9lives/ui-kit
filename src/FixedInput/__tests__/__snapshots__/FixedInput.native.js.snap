// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FixedInput.native renders a snapshot 1`] = `""`;

exports[`FixedInput.native renders a snapshot with show 1`] = `
<Component
  hardwareAccelerated={false}
  transparent={true}
  visible={true}
>
  <KeyboardAvoidingView
    behavior="height"
    enabled={true}
    keyboardVerticalOffset={0}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <TouchableWithoutFeedback>
      <Styled(View)
        align=""
        direction="column"
        grow={1}
        margin={0}
        padding={0}
        style={
          Object {
            "backgroundColor": "transparent",
          }
        }
      />
    </TouchableWithoutFeedback>
    <Styled(Styled(View))>
      <Styled(Styled(View))
        align="center"
      >
        <Styled(View)
          align=""
          direction="row"
          margin={0}
          padding={0}
        >
          <Styled(Text)
            alignSelf="auto"
            color="#191919"
            margin={0}
            numberOfLines={1}
            size={16}
            style={
              Object {
                "flex": 1,
              }
            }
          >
            Label
          </Styled(Text)>
          <Styled(Styled(View))>
            <TouchableOpacity
              activeOpacity={0.2}
              onPress={[Function]}
            >
              <Styled(Text)
                alignSelf="auto"
                color="#191919"
                margin={0}
                size={16}
              >
                Save
              </Styled(Text)>
            </TouchableOpacity>
          </Styled(Styled(View))>
        </Styled(View)>
      </Styled(Styled(View))>
      <Styled(Styled(View))>
        <Styled(TextInput)
          autoFocus={true}
          onChangeText={[Function]}
          placeholder="Placeholder"
        />
      </Styled(Styled(View))>
    </Styled(Styled(View))>
  </KeyboardAvoidingView>
</Component>
`;

exports[`FixedInput.native renders a snapshot with show and label 1`] = `
<Component
  hardwareAccelerated={false}
  transparent={true}
  visible={true}
>
  <KeyboardAvoidingView
    behavior="height"
    enabled={true}
    keyboardVerticalOffset={0}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <TouchableWithoutFeedback>
      <Styled(View)
        align=""
        direction="column"
        grow={1}
        margin={0}
        padding={0}
        style={
          Object {
            "backgroundColor": "transparent",
          }
        }
      />
    </TouchableWithoutFeedback>
    <Styled(Styled(View))>
      <Styled(Styled(View))
        align="center"
      >
        <Styled(View)
          align=""
          direction="row"
          margin={0}
          padding={0}
        >
          <Styled(Text)
            alignSelf="auto"
            color="#191919"
            margin={0}
            numberOfLines={1}
            size={16}
            style={
              Object {
                "flex": 1,
              }
            }
          >
            MyLabel
          </Styled(Text)>
          <Styled(Styled(View))>
            <TouchableOpacity
              activeOpacity={0.2}
              onPress={[Function]}
            >
              <Styled(Text)
                alignSelf="auto"
                color="#191919"
                margin={0}
                size={16}
              >
                Save
              </Styled(Text)>
            </TouchableOpacity>
          </Styled(Styled(View))>
        </Styled(View)>
      </Styled(Styled(View))>
      <Styled(Styled(View))>
        <Styled(TextInput)
          autoFocus={true}
          onChangeText={[Function]}
          placeholder="Placeholder"
        />
      </Styled(Styled(View))>
    </Styled(Styled(View))>
  </KeyboardAvoidingView>
</Component>
`;

exports[`FixedInput.native renders a snapshot with show and multiline 1`] = `
<Component
  hardwareAccelerated={false}
  transparent={true}
  visible={true}
>
  <KeyboardAvoidingView
    behavior="height"
    enabled={true}
    keyboardVerticalOffset={0}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <TouchableWithoutFeedback>
      <Styled(View)
        align=""
        direction="column"
        grow={1}
        margin={0}
        padding={0}
        style={
          Object {
            "backgroundColor": "transparent",
          }
        }
      />
    </TouchableWithoutFeedback>
    <Styled(Styled(View))>
      <Styled(Styled(View))
        align="center"
      >
        <Styled(View)
          align=""
          direction="row"
          margin={0}
          padding={0}
        >
          <Styled(Text)
            alignSelf="auto"
            color="#191919"
            margin={0}
            numberOfLines={1}
            size={16}
            style={
              Object {
                "flex": 1,
              }
            }
          >
            Label
          </Styled(Text)>
          <Styled(Styled(View))>
            <TouchableOpacity
              activeOpacity={0.2}
              onPress={[Function]}
            >
              <Styled(Text)
                alignSelf="auto"
                color="#191919"
                margin={0}
                size={16}
              >
                Save
              </Styled(Text)>
            </TouchableOpacity>
          </Styled(Styled(View))>
        </Styled(View)>
      </Styled(Styled(View))>
      <Styled(Styled(View))>
        <Styled(TextInput)
          autoFocus={true}
          multiline={true}
          onChangeText={[Function]}
          placeholder="Placeholder"
        />
      </Styled(Styled(View))>
    </Styled(Styled(View))>
  </KeyboardAvoidingView>
</Component>
`;

exports[`FixedInput.native renders a snapshot with show and onRequestDismiss 1`] = `
<Component
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  transparent={true}
  visible={true}
>
  <KeyboardAvoidingView
    behavior="height"
    enabled={true}
    keyboardVerticalOffset={0}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <TouchableWithoutFeedback
      onPress={[Function]}
    >
      <Styled(View)
        align=""
        direction="column"
        grow={1}
        margin={0}
        padding={0}
        style={
          Object {
            "backgroundColor": "transparent",
          }
        }
      />
    </TouchableWithoutFeedback>
    <Styled(Styled(View))>
      <Styled(Styled(View))
        align="center"
      >
        <Styled(View)
          align=""
          direction="row"
          margin={0}
          padding={0}
        >
          <Styled(Text)
            alignSelf="auto"
            color="#191919"
            margin={0}
            numberOfLines={1}
            size={16}
            style={
              Object {
                "flex": 1,
              }
            }
          >
            Label
          </Styled(Text)>
          <Styled(Styled(View))>
            <TouchableOpacity
              activeOpacity={0.2}
              onPress={[Function]}
            >
              <Styled(Text)
                alignSelf="auto"
                color="#191919"
                margin={0}
                size={16}
              >
                Save
              </Styled(Text)>
            </TouchableOpacity>
          </Styled(Styled(View))>
        </Styled(View)>
      </Styled(Styled(View))>
      <Styled(Styled(View))>
        <Styled(TextInput)
          autoFocus={true}
          onChangeText={[Function]}
          placeholder="Placeholder"
        />
      </Styled(Styled(View))>
    </Styled(Styled(View))>
  </KeyboardAvoidingView>
</Component>
`;

exports[`FixedInput.native renders a snapshot with show and saveText 1`] = `
<Component
  hardwareAccelerated={false}
  transparent={true}
  visible={true}
>
  <KeyboardAvoidingView
    behavior="height"
    enabled={true}
    keyboardVerticalOffset={0}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <TouchableWithoutFeedback>
      <Styled(View)
        align=""
        direction="column"
        grow={1}
        margin={0}
        padding={0}
        style={
          Object {
            "backgroundColor": "transparent",
          }
        }
      />
    </TouchableWithoutFeedback>
    <Styled(Styled(View))>
      <Styled(Styled(View))
        align="center"
      >
        <Styled(View)
          align=""
          direction="row"
          margin={0}
          padding={0}
        >
          <Styled(Text)
            alignSelf="auto"
            color="#191919"
            margin={0}
            numberOfLines={1}
            size={16}
            style={
              Object {
                "flex": 1,
              }
            }
          >
            Label
          </Styled(Text)>
          <Styled(Styled(View))>
            <TouchableOpacity
              activeOpacity={0.2}
              onPress={[Function]}
            >
              <Styled(Text)
                alignSelf="auto"
                color="#191919"
                margin={0}
                size={16}
              >
                MySaveText
              </Styled(Text)>
            </TouchableOpacity>
          </Styled(Styled(View))>
        </Styled(View)>
      </Styled(Styled(View))>
      <Styled(Styled(View))>
        <Styled(TextInput)
          autoFocus={true}
          onChangeText={[Function]}
          placeholder="Placeholder"
        />
      </Styled(Styled(View))>
    </Styled(Styled(View))>
  </KeyboardAvoidingView>
</Component>
`;
